zetaclient/evm/evm_client.go

```diff
@@ -649,20 +642,14 @@ func (ob *ChainClient) WatchOutTx() {
 				var receipt *ethtypes.Receipt
 				var transaction *ethtypes.Transaction
 				for _, txHash := range tracker.HashList {
-					select {
-					case <-outTimeout:
-						ob.logger.OutTx.Warn().Msgf("WatchOutTx: timeout on chain %d nonce %d", ob.chain.ChainId, nonceInt)
-						break TRACKERLOOP
-					default:
-						if recpt, tx, ok := ob.checkConfirmedTx(txHash.TxHash, nonceInt); ok {
-							txCount++
-							receipt = recpt
-							transaction = tx
-							ob.logger.OutTx.Info().Msgf("WatchOutTx: confirmed outTx %s for chain %d nonce %d", txHash.TxHash, ob.chain.ChainId, nonceInt)
-							if txCount > 1 {
-								ob.logger.OutTx.Error().Msgf(
-									"WatchOutTx: checkConfirmedTx passed, txCount %d chain %d nonce %d receipt %v transaction %v", txCount, ob.chain.ChainId, nonceInt, receipt, transaction)
-							}
+					if recpt, tx, ok := ob.checkConfirmedTx(txHash.TxHash, nonceInt); ok {
```

Comment:
This will make the logic a bit different as we can't assign the returned receipt directly to variable `receipt` without checking `ok == true`, the e2e test will fail.